src/transaction.js

```diff
@@ -140,18 +143,21 @@ Transaction.prototype.hasWitnesses = function () {
 Transaction.prototype.byteLength = function (stype) {
   let hasWitnesses = this.hasWitnesses()
   let onlyWitnesses = false
+  let hashFull = stype === Transaction.TxSerializeFull
```

Comment:
We don't need the additional `hashFull` variable, use `(hasWitness ? 4 : 0) + // Timestamp` is ok